website documentation for running heron on mesos local locally on mac

yanxz opened the pull request.

maosongfu commented:
Can u also modify the website to make it able to navigate to this page?

maosongfu commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,50 @@
+
+This is a step by step guide to run Heron on Mesos cluster locally.
+
+## Install Heron
```
Change from "install heron" to "Get heron with Mesos Scheduler implementation" or someting like it. Otherwise this seems like a little duplicated with section "Compile Heron" below.

Or we maybe move this section to "Compile heron" and merge them together.

maosongfu commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,50 @@
+
+## Compile Heron
+
+Run the command to build and install Heron to pick up all the configuration changes:
```
When submitting a topology, people can also specify the customized config path. The getting-started webpage provides an example.

maosongfu commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
````diff
@@ -0,0 +1,50 @@
+
+```
+heron kill mesos ExclamationTopology
+```
````
Can u also run the heron-tracker and heron-ui, then put their running screenshot into this doc?

maosongfu commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,50 @@
+The output shows if the topology has been launched successfully. Check Mesos management console, you should see the topology:
+
+![result page](/img/mesos-management-console-with-topology.png)
+
```
Can u add more doc here? For instance, where is the scheduler-working-directory, how does it look like, how can people use it to debug. Where's the log of a running heron-instance, what's the sample log content?

maosongfu commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,50 @@
+Follow [Installing Mesos on your Mac with Homebrew](https://mesosphere.com/blog/2014/07/07/installing-mesos-on-your-mac-with-homebrew/) to have Mesos installed and running. To confirm Mesos cluster are ready for accepting Heron topologies, make sure you can access Mesos management console [http://localhost:5050](http://localhost:5050) and see activated slaves.
+
+![console page](/img/mesos-management-console.png)
+
```
I remember you came across some issues when tyring to run the mesos cluster. Could u also share them here as potential trouble shooting?

yanxz replied:
I didn't actually launch the mesos cluster :), which I don't think it's worth mentioning.

objmagic commented:
As for making this webpage reachable from main page, you can modify [toc.yaml](https://github.com/twitter/heron/blob/master/website/data/toc.yaml).

maosongfu commented:
The content looks good to me. :+1:

kramasamy commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,130 @@
+This is a step by step guide to run Heron on Mesos cluster locally.
+
+## Get Heron with Mesos Scheduler Implementation
+At the time this article is written, Mesos scheduler is not included in the release version. So you need the latest version of Heron to run it on Mesos. Follow [Compiling](../../../../developers/compiling/compiling) to make sure you can build from the latest version.
```
Do we need 'At the time this article is written....'?  Please remove - we will forget when we make a release.

billonahill replied:
This whole paragraph can be omitted. Assume the docs ship with the version that has support and that people have that version checked out. You can just start the next paragraph with "After compiling, ..."

kramasamy commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,130 @@
+
+## Setting up Apache Mesos Cluster Locally
+
+Follow [Installing Mesos on your Mac with Homebrew](https://mesosphere.com/blog/2014/07/07/installing-mesos-on-your-mac-with-homebrew/) to have Mesos installed and running. To confirm Mesos cluster are ready for accepting Heron topologies, make sure you can access Mesos management console [http://localhost:5050](http://localhost:5050) and see activated slaves.
```
To confirm Mesos cluster ```is``` ready.

kramasamy commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,130 @@
+## Configure Heron
+
+### State Manager
+By default, Heron uses Local File System State Manager on Mesos to manage states. You can modify `heron/config/src/yaml/conf/mesos/statemgr.yaml` to make it use ZooKeeper. For more details please see [Setting up ZooKeeper](../statemanagers/zookeeper).
```
instead of 'to make it use' change it to 'to use'

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,130 @@
+title: Setting up Heron with Mesos Cluster Locally on Mac
+---
+
+This is a step by step guide to run Heron on Mesos cluster locally.
```
on a Mesos cluster

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,130 @@
+## Get Heron with Mesos Scheduler Implementation
+At the time this article is written, Mesos scheduler is not included in the release version. So you need the latest version of Heron to run it on Mesos. Follow [Compiling](../../../../developers/compiling/compiling) to make sure you can build from the latest version.
+
+## Setting up Apache Mesos Cluster Locally
```
Setting up an Apache Mesos Cluster Locally

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,130 @@
+### Scheduler
+Heron needs to know where to load the lib to interact with Mesos. Change the config `heron.mesos.native.library.path` in `heron/config/src/yaml/conf/mesos/scheduler.yaml` to the library path of your Mesos. If you installed Mesos through `brew`, the library path should be `/usr/local/Cellar/mesos/your_mesos_version/lib`.
+
+> Mesos only offers interface in C++ library, which is not portable across platforms. So Heron has to ask you for the path of the library, which is able to run in your platform. 
```
"Mesos only offers a C++ interface, which is not portable across platforms."

The next sentence is redundant with the first of this section stating that heron needs to know the path, so we can omit. Also best to not talk about what Heron asks for, but to use a less conversational tone. "Heron requires foo. Configure the foo by doing X" for example.

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,130 @@
+
+## Compile Heron
+
+Run the command to build and install Heron:
```
... to install the Heron client.

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
````diff
@@ -0,0 +1,130 @@
+```bash
+bazel run --config=darwin --verbose_failures -- scripts/packages:heron-client-install.sh --user
+```
+This will build everything into `$HOME/.heron`, including all the configurations. In future releases Mesos Scheduler will be included and you will not need to rebuild Heron from source. Then you can add `--config-path=your_conf_path` to your `heron submit` command to tell Heron where to find your config files (default value is `$HOME/.heron/conf`). 
````
Instead of speaking about the future, speak as if it's already here and assume that these docs won't go live until after the code release.

billonahill replied:
please keep line lengths < 100 to make commenting and reading in an ide easier.

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,130 @@
+
+## Run Topology in Mesos
+
+After setting up Heron and Mesos, let's submit a topology to test if everything works right, Run:
```
omit: let's
s/, Run:/./g

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,130 @@
+* `heron-conf`: configurations used to launch the topology.
+* `log-files`: directory containing Mesos scheduler's log.
+
+You can look at the log file to see whether the launch succeeded. If it succeeded, at the end of the log file you'll see:
```
s/You can look at the log file to see/The log file will show/g

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
````diff
@@ -0,0 +1,130 @@
+[2016-07-25 22:15:47 -0700] com.twitter.heron.scheduler.mesos.framework.MesosFramework INFO:  Task with id 'container_1_1469510147073:0' RUNNING
+``` 
+
+If the launch fails, you will see the error message. E.g. if the Heron can't find Mesos library in the given path, it will complain:
````
s/you will see the error message/an error message will be included/g

s/E.g. if the Heron can't find Mesos library in the given path, it will complain:/For example, if the Mesos library isn't found in the configured location, the following exception will occur./g

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,130 @@
+
+## Mesos Management Console
+
+Another way to check your topology is running is to look at the Mesos management console, if the topology was launched successfully you should see two containers of the topology running:
```
In general we want to speak more objectively without reference to 'you' and 'your'.

Another way to check that the topology is running is to look at the Mesos management console. If it was launched successfully, two containers will be running.

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,130 @@
+
+![result page](/img/mesos-management-console-with-topology.png)
+
+They are containers of topology master and heron container. If you click the `sandbox` on the right side, you can access the logs of these processes. E.g. the sandbox of heron container looks as below:
```
omit: They are containers of topology master and heron container.

To view the process logs, click the `sandbox` on the right side. The sandbox of the heron container is shown below.

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,130 @@
+
+![container-container-sandbox](/img/container-container-sandbox.png)
+
+If you go to `log-files` directory, you will see application and GC log of the processes run in this container:
```
The `log-files` directory includes the application and GC log of the processes running in this container.

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,130 @@
+
+![container-log-files](/img/container-log-files.png)
+
+If you click `container_1_exclaim1_1.log.0`, you will see the bolt log of the ExclamationTopology:
```
The bolt log of the ExclamationTopology is `container_1_exclaim1_1.log.0`.

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,130 @@
+
+## Heron UI
+
+If you have Heron tools installed (see [Quick Start Guide](../../../../getting-started)), you can also monitor topology through `heron-ui`. First you need to start the `heron-tracker`. To start the tracker, modify the value of `statemgrs.rootpath` in `$HOME/.herontools/conf/heron_tracker.yaml` to `$HOME/.herondata/repository/state/mesos`. This tells Heron tracker to go to `mesos` directory (defau ...
```
Install Heron tools to monitor the topology with the heron-ui, which required the heron-tracker. Configure ... before starting tracker. This configuration sets the location of the state manager root path.

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
````diff
@@ -0,0 +1,130 @@
+... Using tracker url: http://localhost:8888
+```
+
+Now go to the UI ([http://localhost:8889](http://localhost:8889)) and you'll see your topology:
````
Go to the UI to see the topology.

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,130 @@
+
+![mesos-local-heron-ui](/img/mesos-local-heron-ui.png)
+
+Click on the topology, you'll see its metrics:
```
To see the metrics, click on the topology.

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,130 @@
+
+![mesos-local-heron-ui-more](/img/mesos-local-heron-ui-more.png)
+
+If you click on the `job` button, you'll enter the Mesos Management Console page:
```
To enter the Mesos Management Console page, click the job button.

yanxz commented:
Thanks for the detailed suggestions @billonahill @kramasamy, addressed all the comments. @billonahill deleted the contents about compiling Heron as this doc will be released with the Mesos Scheduler. User can use the binaries then.

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,119 @@
+
+## Setting up an Apache Mesos Cluster Locally
+
+Follow [Installing Mesos on your Mac with Homebrew](https://mesosphere.com/blog/2014/07/07/installing-mesos-on-your-mac-with-homebrew/) to have Mesos installed and running. To confirm Mesos cluster is ready for accepting Heron topologies, make sure you can access Mesos management console [http://localhost:5050](http://localhost:5050) and see activated slaves.
```
s/to have Mesos installed and running/to install and run Mesos/

s/make sure you can access/access the/

Also please limit line lenghts here and elsewhere to 100 chars to make commenting and reading in an IDE easier.

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,119 @@
+## Configure Heron
+
+### State Manager
+By default, Heron uses Local File System State Manager on Mesos to manage states. You can modify `$HOME/.heron/conf/mesos/statemgr.yaml` to use ZooKeeper. For more details please see [Setting up ZooKeeper](../statemanagers/zookeeper).
```
s/You can modify/Modify/

omit: please

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,119 @@
+By default, Heron uses Local File System State Manager on Mesos to manage states. You can modify `$HOME/.heron/conf/mesos/statemgr.yaml` to use ZooKeeper. For more details please see [Setting up ZooKeeper](../statemanagers/zookeeper).
+
+### Scheduler
+Heron needs to know where to load the lib to interact with Mesos. Change the config `heron.mesos.native.library.path` in `$HOME/.heron/conf/mesos/scheduler.yaml` to the library path of your Mesos. If Mesos is installed through `brew`, the library path should be `/usr/local/Cellar/mesos/your_mesos_version/lib`.
```
s/your Mesos/the Meso install/

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,119 @@
+### Scheduler
+Heron needs to know where to load the lib to interact with Mesos. Change the config `heron.mesos.native.library.path` in `$HOME/.heron/conf/mesos/scheduler.yaml` to the library path of your Mesos. If Mesos is installed through `brew`, the library path should be `/usr/local/Cellar/mesos/your_mesos_version/lib`.
+
+> Mesos only offers interface in C++ library, which is not portable across platforms.
```
Mesos only offers a C++ interface,

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,119 @@
+
+## Run Topology in Mesos
+
+After setting up Heron and Mesos, submit a topology to test if everything works right. By default this command loads the config in `$HOME/.heron/conf`, add `--config-path=your_conf_path` to change the config path.
```
s/to test if everything works right/using the following command/

s/, add/. Add/

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
````diff
@@ -0,0 +1,119 @@
+heron submit mesos --verbose ~/.heron/examples/heron-examples.jar com.twitter.heron.examples.ExclamationTopology ExclamationTopology
+```
+
+If the topology is submitted successfully, you will see:
````
The following will be displayed upon a successful submit.

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
````diff
@@ -0,0 +1,119 @@
+INFO: Elapsed time: 4.114s.
+``` 
+
+Note that this doesn't necessarily mean it's successfully launched in Mesos, to check that go to the working directory as shown in the output. You will see:
````
s/it's/the topology/

s/, to check that go the/. To verify check the

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
```diff
@@ -0,0 +1,119 @@
+
+## Heron UI
+
+Install Heron tools to monitor the topology with the `heron-ui` (see [Quick Start Guide](../../../../getting-started)). Configure the value of `statemgrs.rootpath` in `$HOME/.herontools/conf/heron_tracker.yaml` to `$HOME/.herondata/repository/state/mesos` before starting the tracker. This configuration sets the location of the state manager root path. Start tracker and UI.
```
Start tracker and the UI

billonahill commented on website/content/docs/operators/deployment/schedulers/mesos-local-mac.md:
````diff
@@ -0,0 +1,119 @@
+... Using tracker url: http://localhost:8888
+```
+
+Go to the UI ([http://localhost:8889](http://localhost:8889)) to see the topology.
````
Go to the UI at http:...

yanxz commented:
@billonahill sorry forgot the char number issue in my last commit. Addressed all the comments.

objmagic commented:
👍

billonahill commented:
Thanks @yanxz looks good! Would you please accept the CLA:
http://twitter.github.io/heron/docs/contributors/community#submitting-a-patch

yanxz commented:
@billonahill I've accepted the CLA.

kramasamy commented:
Thanks @yanxz
